3rdParty/rocksdb/v5.1.4/include/rocksdb/env_encryption.h

```diff
@@ -125,27 +132,65 @@ class EncryptionProvider {
  public:
     virtual ~EncryptionProvider() {};
 
+    // GetPrefixLength returns the length of the prefix that is added to every file
+    // and used for storing encryption options.
+    // For optimal performance, the prefix length should be a multiple of 
+    // the a page size.
+    virtual size_t GetPrefixLength() = 0;
+
+    // CreateNewPrefix initialized an allocated block of prefix memory 
+    // for a new file.
+    virtual Status CreateNewPrefix(const std::string& fname, char *prefix, size_t prefixLength) = 0;
+
     // CreateCipherStream creates a block access cipher stream for a file given
     // given name and options.
     virtual Status CreateCipherStream(const std::string& fname, const EnvOptions& options,
-                                 unique_ptr<BlockAccessCipherStream>* result) = 0;
+      Slice& prefix, unique_ptr<BlockAccessCipherStream>* result) = 0;
 };
 
 // This encryption provider uses a CTR cipher stream, with a given block cipher 
 // and IV.
+//
+// Note: This is a possible implementation of EncryptionProvider, 
+// it is considered suitable for use, provided a safe BlockCipher is used.
 class CTREncryptionProvider : public EncryptionProvider {
     private:
       BlockCipher& cipher_;
-      const char* iv_;
+    protected:
+      const static size_t defaultPrefixLength = 4096;
```

Comment:
I fear that if we make that configurable people will be much more likely to screw things up.
We chose the 4K size because it is a suitable page size on pretty much every platform we can think of. 

Why should this be make configurable?